### PR TITLE
Asyncio Transport Closed Explicitly, Fixes for Dumping Address Space

### PR DESCRIPTION
**Pull request by zerox1212**
Fixes for issues discussed at the bottom of https://github.com/FreeOpcUa/python-opcua/issues/60 . Please review changes in binary_server_asyncio.py closely as this is the best solution I was able to come up with. Especially the while loop in stop(), this is the only way I could make sure the transport was really closed before the server is closed. Without this there seems to be a race condition.

I also added the address space dump methods back into the high level server because it seems like it could be a nice feature for people that want to back up their address space easily.

TLDR:
can't dump address space without cleaning up client connections

### CODE REVIEW

File: opcua/server/address_space.py [hunk 1]
```diff
@@ -428,8 +428,11 @@ def dump(self, path):
         """
         dump address space as binary to file
         """
-        with open(path, 'wb') as f:
-            pickle.dump(self._nodes, f, pickle.HIGHEST_PROTOCOL)
+        try:
+            with open(path, 'wb') as f:
+                pickle.dump(self._nodes, f, pickle.HIGHEST_PROTOCOL)
+        except TypeError:
+            self.logger.info("Failed to dump address space")
```
Comment (oroulet) [inline opcua/server/address_space.py]: do not catch things here, it shoulb be catch at higher level

File: opcua/server/binary_server_asyncio.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def __init__(self, internal_server, hostname, port):
         self.loop = internal_server.loop
         self._server = None
         self._policies = []
+        self.shutdown = [False]  # needs to be immutable list for referencing reasons
```
Comment (oroulet) [inline opcua/server/binary_server_asyncio.py]: no way :-) This is clearly impossible that this is required

**Comment by oroulet** [comment]
seems to break tests

**Comment by zerox1212** [comment]
Well I tried. I'm still new at programming. I will cancel this pull request and fix address space dumping only in my local code.

However, I still think transport needs to be closed explicitly when OPC UA server shuts down.

Thanks for checking it.

**Comment by oroulet** [comment]
No problem :-) you're attacking the worst thing in programming: multi threading.... I have been doing that kind of things in 15 years and make mistakes all the time...
